geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoVisibilityRowEncoder.scala

```diff
@@ -88,18 +89,27 @@ class KryoVisibilityRowEncoder extends RowEncodingIterator {
       val input = KryoFeatureDeserialization.getInput(bytes, 1, bytes.length - 1) // skip the version byte
       val metadata = Metadata(input) // read count, size, etc
       // Column qualifiers tell us which attributes are intended to be populated given visibilities
-      keys.get(i).getColumnQualifier.getBytes.foreach { unsigned =>
+      val key = keys.get(i)
+      key.getColumnQualifier.getBytes.foreach { unsigned =>
         val index = java.lang.Byte.toUnsignedInt(unsigned)
         if(!metadata.nulls.contains(index)) {
           val pos = metadata.setPosition(index)
           val len = metadata.setPosition(index + 1) - pos
           attributes(index) = (bytes, 4 + pos, len) // pos is relative to first byte of attribute offsets, see length calc
+          attributeVis(index) = key.getColumnVisibility.toString
           length += len
         }
       }
       i += 1
     }
 
+    val attributeVisString = attributeVis.mkString(",")
```

Comment:
if you can't see all the attributes, this would end up with 'null's in the string. maybe we should replace those with empty strings?